src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/MarketsProcessor.kt

```diff
@@ -1,25 +1,109 @@
 package exchange.dydx.abacus.processor.markets
 
 import exchange.dydx.abacus.processor.base.BaseProcessor
+import exchange.dydx.abacus.processor.base.BaseProcessorProtocol
 import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.state.internalstate.InternalMarketState
+import exchange.dydx.abacus.state.internalstate.InternalMarketSummaryState
 import exchange.dydx.abacus.utils.Logger
 import exchange.dydx.abacus.utils.mutable
 import exchange.dydx.abacus.utils.safeSet
+import indexer.models.IndexerCompositeMarketObject
+import indexer.models.IndexerWsMarketUpdateResponse
+
+internal interface MarketsProcessorProtocol : BaseProcessorProtocol {
+    fun processSubscribed(
+        existing: InternalMarketSummaryState,
+        content: Map<String, IndexerCompositeMarketObject>?
+    ): InternalMarketSummaryState
+
+    fun processChannelData(
+        existing: InternalMarketSummaryState,
+        content: IndexerWsMarketUpdateResponse?,
+    ): InternalMarketSummaryState
+
+    fun processChannelBatchData(
+        existing: InternalMarketSummaryState,
+        content: List<IndexerWsMarketUpdateResponse>?,
+    ): InternalMarketSummaryState
+}
 
-@Suppress("UNCHECKED_CAST")
 internal class MarketsProcessor(
     parser: ParserProtocol,
-    calculateSparklines: Boolean
-) : BaseProcessor(parser) {
-    private val marketProcessor = MarketProcessor(parser, calculateSparklines)
+    calculateSparklines: Boolean,
+    private val marketProcessor: MarketProcessorProtocol = MarketProcessor(parser, calculateSparklines)
+) : BaseProcessor(parser), MarketsProcessorProtocol {
+    private val marketProcessorDeprecated: MarketProcessor? = marketProcessor as? MarketProcessor
 
     internal var groupingMultiplier: Int
-        get() = marketProcessor.groupingMultiplier
+        get() = marketProcessorDeprecated!!.groupingMultiplier
         set(value) {
-            marketProcessor.groupingMultiplier = value
+            marketProcessorDeprecated?.groupingMultiplier = value
```

Comment:
why the `?.` safe accessor here but the `!!.` in the `get`?

Reply:
Either is fine.... it's there just to keep the legacy code happy.